stl/inc/sstream

```diff
@@ -100,6 +151,68 @@ public:
         _Init(_Newstr.c_str(), _Newstr.size(), _Mystate);
     }
 
+#if _HAS_CXX20
+    template <class _Alloc2, enable_if_t<_Is_allocator<_Alloc2>::value, int> = 0>
+    _NODISCARD basic_string<_Elem, _Traits, _Alloc2> str(const _Alloc2& _Al) const {
+        return basic_string<_Elem, _Traits, _Alloc2>{view(), _Al};
+    }
+
+    // The buffer cannot be moved to string directly, because
+    // the buffer maybe already full, and the terminating char is not \0.
+    // In that case, copy the string as usual.
+    _NODISCARD _Mystr str() && {
+        _Mystr _Result(_Al);
+        if (!(_Mystate & _Constant) && _Mysb::pptr()) { // writable
+            const auto _Base   = _Mysb::pbase();
+            const auto _Ptr    = _Mysb::pptr();
+            const auto _Eptr   = _Mysb::epptr();
+            const auto _Mysize = static_cast<_Mysize_type>((_STD max)(_Ptr, _Seekhigh) - _Base);
+            if (_Ptr == _Eptr
+                && !_Traits::eq(*(_Ptr - 1), _Elem())) { // the buffer is full and terminating char is not \0
+                _Result.assign(_Base, _Mysize);
+            } else {
+                _Result._Move_assign_from_buffer(_Base, _Mysize, static_cast<_Mysize_type>(_Eptr - _Base));
+                _Mystate &= ~_Allocated;
+            }
+        } else if (!(_Mystate & _Noread) && _Mysb::gptr()) { // readable
+            const auto _Base   = _Mysb::eback();
+            const auto _Eptr   = _Mysb::egptr();
+            const auto _Mysize = static_cast<_Mysize_type>(_Eptr - _Base);
+            if (!_Traits::eq(*(_Eptr - 1), _Elem())) { // the buffer is full and terminating char is not \0
+                _Result.assign(_Base, _Mysize);
+            } else {
+                _Result._Move_assign_from_buffer(_Base, _Mysize, _Mysize);
+                _Mystate &= ~_Allocated;
+            }
+        }
+        _Tidy();
+        return _Result;
+    }
+
+    _NODISCARD _Mystr_view view() const noexcept {
+        auto _Pair = _View_pair();

```

Comment:
This could be `const`. In new code, we try to mark value parameters and local variables as `const` whenever doing so doesn't inhibit optimizations or interfere with anything.